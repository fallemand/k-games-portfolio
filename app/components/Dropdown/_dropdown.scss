@import '../../styles/variables';
@import './dropdown__small';

@media ( min-width: $breakpoint-desktop ) {
  @import './dropdown__medium';
}
