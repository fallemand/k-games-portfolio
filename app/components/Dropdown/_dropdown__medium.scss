.dropdown {
  min-width: 150px;
  margin: 0 10px;
  width: auto;
  display: inline-block;
}
