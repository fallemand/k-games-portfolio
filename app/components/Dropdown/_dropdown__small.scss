.dropdown {
  height: 30px;
  border-radius: 2px;
  display: block;
  width: 100%;
  margin: 10px 0;
  text-align-last: center;
  border-color: $border-color;
}
