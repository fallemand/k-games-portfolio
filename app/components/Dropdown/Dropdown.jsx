import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import './_dropdown.scss';

class Dropdown extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const { value } = event.target;
    const { onChange } = this.props;
    onChange(value);
  }

  render() {
    const {
      className, name, title, options, value,
    } = this.props;
    return (
      <select className={classnames(className, 'dropdown')} name={name} onChange={this.handleChange}>
        {title && <option value="" disabled selected={!value}>{title}</option>}
        {
          options.map(option => (
            <option
              key={option.value}
              value={option.value}
              disabled={option.disabled}
              selected={option.value === value}
            >
              {option.label}
            </option>
          ))
        }
      </select>
    );
  }
}
Dropdown.propTypes = {
  className: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  title: PropTypes.string,
  options: PropTypes.array.isRequired,
};

Dropdown.defaultProps = {
  className: null,
  title: null,
  value: '',
};

export default Dropdown;
