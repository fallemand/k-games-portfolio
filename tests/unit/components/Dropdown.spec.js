import React from 'react';
import Dropdown from '../../../app/components/Dropdown';

describe('Dropdown', () => {
  const props = {
    name: '__NAME__',
    title: '__TITLE__',
    className: '__CLASSNAME__',
    options: [
      { label: 'LABEL1', value: 'VALUE1' },
      { label: 'LABEL2', value: 'VALUE2', disabled: true },
      { label: 'LABEL3', value: 'VALUE3' },
    ],
    onChange: jest.fn(),
  };

  it('render with props', () => {
    const component = shallow(<Dropdown {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('render with selected value', () => {
    const component = shallow(<Dropdown {...props} value="VALUE3" />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('on change value should call onChange', () => {
    const component = shallow(<Dropdown {...props} />);
    component.find('select').simulate('change', { target: { value: 'VALUE1' } });
    expect(props.onChange).toHaveBeenCalledWith('VALUE1');
  });
});
