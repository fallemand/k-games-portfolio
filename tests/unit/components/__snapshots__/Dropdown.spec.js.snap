// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown render with props 1`] = `
"<select className='__CLASSNAME__ dropdown' name='__NAME__' onChange={[Function: bound handleChange]}>
  <option value='' disabled={true} selected={true}>
    __TITLE__
  </option>
  <option value='VALUE1' disabled={[undefined]} selected={false}>
    LABEL1
  </option>
  <option value='VALUE2' disabled={true} selected={false}>
    LABEL2
  </option>
  <option value='VALUE3' disabled={[undefined]} selected={false}>
    LABEL3
  </option>
</select>"
`;

exports[`Dropdown render with selected value 1`] = `
"<select className='__CLASSNAME__ dropdown' name='__NAME__' onChange={[Function: bound handleChange]}>
  <option value='' disabled={true} selected={false}>
    __TITLE__
  </option>
  <option value='VALUE1' disabled={[undefined]} selected={false}>
    LABEL1
  </option>
  <option value='VALUE2' disabled={true} selected={false}>
    LABEL2
  </option>
  <option value='VALUE3' disabled={[undefined]} selected={true}>
    LABEL3
  </option>
</select>"
`;
